tinyscheme fails to compile with GCC 15 / C23: isascii undeclared

## Problem

Building gerbv with GCC 15 on Linux fails with:

```
thirdparty/tinyscheme/scheme.c: In function 'Cisalpha':
thirdparty/tinyscheme/scheme.c:269:44: error: implicit declaration of function 'isascii' [-Wimplicit-function-declaration]
  269 | static INLINE int Cisalpha(int c) { return isascii(c) && isalpha(c); }
      |                                            ^~~~~~~
```

`isascii` is a POSIX function, not part of the C standard. The project sets `CMAKE_C_STANDARD 23` with `CMAKE_C_EXTENSIONS OFF`, so GCC compiles with `-std=c23` which doesn't expose POSIX extensions. In C23, implicit function declarations are a hard error.

## Root cause

The `compilation-flags` INTERFACE target defines `_GNU_SOURCE` on Linux (which exposes `isascii` from `<ctype.h>`), but `tinyscheme_bundled` links only `config` — it never gets `_GNU_SOURCE`.

## Fix

Define `_GNU_SOURCE` on the `tinyscheme_bundled` target for Linux, matching the existing pattern in `compilation-flags`.

--- a/thirdparty/tinyscheme/GERBV_PATCHES
+++ b/thirdparty/tinyscheme/GERBV_PATCHES
@@ -5,3 +5,6 @@ License: BSD 3-Clause (see COPYING)
 
 Bundled with gerbv for the project file parser (project.c).
 Used as an embedded interpreter — not built standalone.
+
+The following known patches added so far:
+1. The isascii() function in scheme.c replaced by our own local version

--- a/thirdparty/tinyscheme/scheme.c
+++ b/thirdparty/tinyscheme/scheme.c
@@ -266,11 +266,12 @@ INTERFACE INLINE void setimmutable(pointer p) { typeflag(p) |= T_IMMUTABLE; }
 #define cddddr(p)        cdr(cdr(cdr(cdr(p))))
 
 #if USE_CHAR_CLASSIFIERS
-static INLINE int Cisalpha(int c) { return isascii(c) && isalpha(c); }
-static INLINE int Cisdigit(int c) { return isascii(c) && isdigit(c); }
-static INLINE int Cisspace(int c) { return isascii(c) && isspace(c); }
-static INLINE int Cisupper(int c) { return isascii(c) && isupper(c); }
-static INLINE int Cislower(int c) { return isascii(c) && islower(c); }
+static INLINE int isascii_local(int c) { return (c >= 0) && (c < 128); }
+static INLINE int Cisalpha(int c) { return isascii_local(c) && isalpha(c); }
+static INLINE int Cisdigit(int c) { return isascii_local(c) && isdigit(c); }
+static INLINE int Cisspace(int c) { return isascii_local(c) && isspace(c); }
+static INLINE int Cisupper(int c) { return isascii_local(c) && isupper(c); }
+static INLINE int Cislower(int c) { return isascii_local(c) && islower(c); }
 #endif
 
 #if USE_ASCII_NAMES

